libr/anal/cc.c

```diff
@@ -1,354 +1,39 @@
-/* radare - LGPL - Copyright 2011-2012 - pancake */
+/* radare - LGPL - Copyright 2016 - Oddcoder */
 
-// moved from core/cmd.c // the loop that does the funny trick ..
-#if 0
-NOTES
-=====
-- RAnalCC must be defined in every function.. maybe in xrefs? or per-opcode cache?
-  - hashtable with offset-opcode must be faster to index than to recalc every time
-- Interrupts (SWI) will use always a predefined type.. this must be rethinked
-#endif
+/* Universal calling convention implementation based on sdb */
 
 #include <r_anal.h>
-
-R_API RAnalCC* r_anal_cc_new () {
-	RAnalCC *cc = R_NEW0 (RAnalCC);
-	if (!cc) return NULL;
-	r_anal_cc_init (cc);
-	return cc;
-}
-
-R_API void r_anal_cc_init (RAnalCC *cc) {
-	memset (cc, 0, sizeof (RAnalCC));
-}
-
-R_API int r_anal_cc_str2type (const char *str) {
-	if (!strcmp (str, "cdecl")) return R_ANAL_CC_TYPE_CDECL;
-	if (!strcmp (str, "stdcall")) return R_ANAL_CC_TYPE_STDCALL;
-	if (!strcmp (str, "fastcall")) return R_ANAL_CC_TYPE_FASTCALL;
-	if (!strcmp (str, "pascal")) return R_ANAL_CC_TYPE_PASCAL;
-	if (!strcmp (str, "winapi")) return R_ANAL_CC_TYPE_WINAPI;
-	if (!strcmp (str, "msfastcall")) return R_ANAL_CC_TYPE_MSFASTCALL;
-	if (!strcmp (str, "bofastcall")) return R_ANAL_CC_TYPE_BOFASTCALL;
-	if (!strcmp (str, "wafastcall")) return R_ANAL_CC_TYPE_WAFASTCALL;
-	if (!strcmp (str, "clarion")) return R_ANAL_CC_TYPE_CLARION;
-	if (!strcmp (str, "safecall")) return R_ANAL_CC_TYPE_SAFECALL;
-	if (!strcmp (str, "sysv")) return R_ANAL_CC_TYPE_SYSV;
-	if (!strcmp (str, "thiscall")) return R_ANAL_CC_TYPE_THISCALL;
-	return -1;
-}
-
-R_API const char *r_anal_cc_type2str(int type) {
-	switch (type) {
-	case R_ANAL_CC_TYPE_CDECL: return "cdecl";
-	case R_ANAL_CC_TYPE_STDCALL: return "stdcall";
-	case R_ANAL_CC_TYPE_FASTCALL: return "fastcall";
-	case R_ANAL_CC_TYPE_PASCAL: return "pascal";
-	case R_ANAL_CC_TYPE_WINAPI: return "winapi";
-	case R_ANAL_CC_TYPE_MSFASTCALL: return "msfastcall";
-	case R_ANAL_CC_TYPE_BOFASTCALL: return "bofastcall";
-	case R_ANAL_CC_TYPE_WAFASTCALL: return "wafastcall";
-	case R_ANAL_CC_TYPE_CLARION: return "clarion";
-	case R_ANAL_CC_TYPE_SAFECALL: return "safecall";
-	case R_ANAL_CC_TYPE_SYSV: return "sysv";
-	case R_ANAL_CC_TYPE_THISCALL: return "thiscall";
+#ifdef DB
```

Comment:
wtf, dont reuse DB more than once per file, just use another define or use the anal->sdb_cc directly.